cpp/src/parquet/arrow/writer.cc

```diff
@@ -291,8 +291,13 @@ class FileWriterImpl : public FileWriter {
         arrow_properties_(std::move(arrow_properties)),
         closed_(false) {
     if (arrow_properties_->use_threads()) {
-      parallel_column_write_contexts_.resize(schema_->num_fields(),
-                                             {pool, arrow_properties_.get()});
```

Comment:
Indeed, for `vector.resize()`, all vector elements are copy constructed from a single object [1], but `ArrowWriteContext` doesn't handle copying shared_ptr data members [2].

I wonder if buffers in ArrowWriteContext supposed to be shared by multiple instances. If not, we can use unique_ptr.

[1] https://godbolt.org/z/h7o3eW378
[2] https://github.com/apache/arrow/blob/master/cpp/src/parquet/properties.h#L957

Reply:
I believe `parquet::AllocateBuffer` returns a `shared_ptr` which might make that difficult.  However, we could probably change that too.